File: src/main/java/net/cakemc/database/file/MemoryFile.java
package net.cakemc.database.file;

import java.nio.file.Path;

/**
 * The type Memory file.
 */
public abstract class MemoryFile {
    /**
     * Gets path.
     *
     * @return the path
     */
    public abstract Path getPath();

    /**
     * Get data byte [ ].
     *
     * @return the byte [ ]
     */
    public abstract byte[] getData();

    /**
     * Sets data.
     *
     * @param data the data
     */
    public abstract void setData(byte[] data);
}

File: src/main/kotlin/net/cakemc/database/file/MemoryFile.kt
package net.cakemc.database.file

import java.nio.file.Path

/**
 * The type Memory file.
 */
abstract class MemoryFile {
    /**
     * Gets path.
     *
     * @return the path
     */
    abstract val path: Path

    /**
     * Get data byte [ ].
     *
     * @return the byte [ ]
     */
    /**
     * Sets data.
     *
     * @param data the data
     */
    abstract var data: ByteArray
}
